prometheus-client-metrics/src/main/scala/org/http4s/client/prometheus/PrometheusClientMetrics.scala

```diff
@@ -0,0 +1,131 @@
+package org.http4s.client.prometheus
+
+import io.prometheus.client._
+import cats.implicits._
+import cats.data.Kleisli
+import cats.effect.Sync
+import org.http4s._
+import org.http4s.client.{Client, DisposableResponse}
+
+object PrometheusClientMetrics {
+
+  private def reportStatus(status: Status): String =
+    status.code match {
+      case hundreds if hundreds < 200 => "1xx"
+      case twohundreds if twohundreds < 300 => "2xx"
+      case threehundreds if threehundreds < 400 => "3xx"
+      case fourhundreds if fourhundreds < 500 => "4xx"
+      case _ => "5xx"
+    }
+
+  /** The value of this key in the request's attributes is used as the value for the destination metric label. */
+  val Destination = AttributeKey[String]
+  val EmptyDestination = ""
+  private def reportDestination(attributes: AttributeMap, destination: String): String =
+    attributes.get(Destination).getOrElse(destination)
+
+  private case class ClientMetrics(
+      destination: String,
+      responseDuration: Histogram,
+      activeRequests: Gauge,
+      responseCounter: Counter,
+      clientErrorsCounter: Counter
+  )
+
+  private def metricsClient[F[_]: Sync](
+      metrics: ClientMetrics,
+      client: Client[F]
+  )(
+      request: Request[F]
+  ): F[DisposableResponse[F]] =
+    for {
+      start <- Sync[F].delay(System.nanoTime())
+      _ <- Sync[F].delay(
+        metrics.activeRequests
+          .labels(reportDestination(request.attributes, metrics.destination))
+          .inc())
+      responseAttempt <- client.open(request).attempt
+      end <- Sync[F].delay(System.nanoTime())
+      result <- responseAttempt.fold(
+        e =>
+          onClientError(request, metrics) *>
+            Sync[F].raiseError[DisposableResponse[F]](e),
+        r => onResponse(request, r.response, start, end, metrics) *> r.pure[F]
+      )
+    } yield result
+
+  private def onClientError[F[_]: Sync](request: Request[F], metrics: ClientMetrics): F[Unit] =
+    Sync[F].delay {
+      //not updating responseDuration or responseCounter, since we did not receive a response
```

Comment:
Maybe add an error counter?

Reply:
`clientErrorsCounter` is incremented a few lines below, would that be enough?